test/e2e/validation_test.go

```diff
@@ -289,13 +288,31 @@ func (tc *testContext) sshSetup() error {
 	return nil
 }
 
-// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
-// by the provided IP.
-func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+// runPowershellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
+// powershell, on the host specified by the provided IP.
+func (tc *testContext) runPowershellSSHJob(name, command, ip string) (string, error) {
+	keyMountDir := "/private-key"
+	sshCommand := []string{"bash", "-c",
+		fmt.Sprintf(
+			// first determine if the host has powershell or cmd as the default shell
+			"PSVERSION_TABLE=$(ssh -o StrictHostKeyChecking=no -i %s %s@%s 'echo ($PSVersionTable)');"+
+				"if [[ \"$PSVERSION_TABLE\" == *\"PSEdition\"* ]]; "+
+				"then CMD_PREFIX=\"\";CMD_SUFFIX=\"\";"+
+				// if powershell is not the default shell, explicitly run the command through powershell
+				"else CMD_PREFIX=\"powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \\\"\";CMD_SUFFIX=\"\\\"\";"+
+				"fi;"+
+				"ssh -o StrictHostKeyChecking=no -i %s %s@%s ${CMD_PREFIX}' %s '${CMD_SUFFIX}",
+			filepath.Join(keyMountDir, secrets.PrivateKeySecretKey), tc.vmUsername(), ip,
+			filepath.Join(keyMountDir, secrets.PrivateKeySecretKey), tc.vmUsername(), ip, command)}
+
+	return tc.runToolsJob(name, sshCommand)
+}
+
+// runTools creates and waits for a Kubernetes job to run. The command provided will be executed on a linux worker,
+// using the tools image.
+func (tc *testContext) runToolsJob(name string, command []string) (string, error) {
```

Comment:
Shouldn't this just be `runJob`? What is `tool` signifying here?

Reply:
That it uses the using the tools image

Reply:
I don't see any reason for the name change given we were always using the tools image.